ckanext/unhcr/blueprints/dataset.py

```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
+
+import logging
+from flask import Blueprint
+from ckan import model
+from ckan.views.dataset import CreateView as DatasetCreateView
+from ckan.views.resource import CreateView as ResourceCreateView
+import ckan.plugins.toolkit as toolkit
+from ckanext.scheming.helpers import scheming_get_dataset_schema
+from ckanext.unhcr import mailer
+from ckanext.unhcr.activity import log_download_activity
+from ckanext.unhcr.utils import resource_is_blocked
+
+
+log = logging.getLogger(__name__)
+
+
+unhcr_dataset_blueprint = Blueprint(
+    'unhcr_dataset',
+    __name__,
+    url_prefix=u'/dataset',
+    url_defaults={u'package_type': u'dataset'},
+)
+
+
+def publish(package_type, dataset_id):
+    if (not hasattr(toolkit.c, "user") or not toolkit.c.user):
+        return toolkit.abort(403, "Forbidden")
+
+    context = {'model': model, 'user': toolkit.c.user}
+    dataset = toolkit.get_action('package_show')(context.copy(), {'id': dataset_id})
+    if len(dataset['resources']) == 0:
+        return toolkit.abort(400, 'Dataset must have one or more resources to publish')
+    dataset = toolkit.get_action('package_patch')(context.copy(), {'id': dataset_id, 'state': 'active'})
+    toolkit.h.flash_success('Dataset "%s" has been published' % dataset['title'])
+    return toolkit.redirect_to('dataset_read', id=dataset['name'])
```

Comment:
Flask routes use dots. Underscores should still work but let's future-proof it whenever we can.
```suggestion
    return toolkit.redirect_to('dataset.read', id=dataset['name'])
```

Reply:
Yep. I've got a note to go through all these properly when I review all the `url_for`/`link_for`/`redirect_to` calls: https://github.com/okfn/ckanext-unhcr/issues/530#issuecomment-826727562